multiqc/modules/ngsderive/ngsderive.py

```diff
@@ -124,37 +112,61 @@ def probe_file_for_dictreader_kwargs(self, f, expected_header_count):
         f.seek(0)
 
         for delim in ["\t", ",", " ", "|"]:
-            if delim in header and len(header.split(delim)) == expected_header_count:
+            if delim in header and len(header.split(delim)) in expected_header_counts:
                 return {"delimiter": delim}
 
         log.warn("Could not easily detect delimiter for file. Trying csv.Sniffer()")
         dialect = csv.Sniffer().sniff(f.read(4096))  # 4096 is the buffer size
         f.seek(0)
         return {"dialect": dialect}
 
-    def parse(self, sample_dict, found_file, subcommand, expected_header_count):
-        kwargs = self.probe_file_for_dictreader_kwargs(io.StringIO(found_file["f"]), expected_header_count)
+    def parse(self, sample_dict, found_file, subcommand, expected_header_counts):
+        kwargs = self.probe_file_for_dictreader_kwargs(io.StringIO(found_file["f"]), expected_header_counts)
         for row in csv.DictReader(io.StringIO(found_file["f"]), **kwargs):
             if not row.get("File"):
                 continue
             sample_name = self.clean_s_name(row.get("File"), found_file)
+            read_group = row.get("ReadGroup")
+            if read_group and read_group != "overall":
+                sample_name += "-" + read_group
+                row["is_read_group"] = True
+            else:
+                row["is_read_group"] = False
```

Comment:
@vladsavelyev At this stage of development, the only subcommands which support read group level data are `strandedness` and `endedness`. That data is not guaranteed to be present, as a user must opt into that level of granularity.

I would like to support two different modes for the `strandedness` barplot. The default would be the current implementation (only one "overall" entry per-sample). The user could then toggle the "Read Group" mode which shows samples + their read group level entries. I'll need to play around with the sorting here to see what would be most useful. I'm not sure ATM.

I don't think we need any toggles for the `endedness` table. I think it's suitable to display all found information by default.

Is there an example elsewhere in the code base I could follow for what I'm describing for `strandedness`? I'm not quite sure the best way to approach the problem.